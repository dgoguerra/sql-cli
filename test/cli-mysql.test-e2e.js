const Knex = require("knex");
const { cliTestSuite } = require("./cli-test-suite");

const TEST_MYSQL_CONN = {
  host: "127.0.0.1",
  port: 23306,
  user: "test",
  password: "Secret123",
  database: "test_db",
  charset: "utf8mb4",
  timezone: "+00:00",
};

cliTestSuite("mysql2", () =>
  Knex({ client: "mysql2", connection: TEST_MYSQL_CONN })
);
