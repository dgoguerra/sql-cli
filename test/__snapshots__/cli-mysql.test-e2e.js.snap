// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI test: mysql2 can connect through ssh 1`] = `
"table   rows bytes
table_1 2    32.8 kB
table_3 0    32.8 kB
table_2 2    16.4 kB

(81.9 kB in 3 tables)
"
`;

exports[`CLI test: mysql2 can connect through ssh 2`] = `
"column     type         nullable default foreign key
id         int unsigned false
field_1    int          true
field_2    varchar(50)  true
created_at datetime     true
updated_at datetime     true

index                 algorithm unique columns
PRIMARY               BTREE     true   id
table_1_field_1_index BTREE     false  field_1
"
`;

exports[`CLI test: mysql2 can connect with an alias 1`] = `
"table   rows bytes
table_1 2    32.8 kB
table_3 0    32.8 kB
table_2 2    16.4 kB

(81.9 kB in 3 tables)
"
`;

exports[`CLI test: mysql2 can diff tables 1`] = `
"column  type                        nullable     default foreign key
id      int unsigned → varchar(255) false
field_1 int → decimal(8,2)          true → false
field_2 varchar(50) → text(65535)   true

index                  algorithm unique columns
table_1_field_1_index  BTREE     false  field_1
table_2_field_1_unique BTREE     true   field_1

Columns: 3x changed, 2x similar (hidden)
Indexes: 1x deleted, 1x created, 1x similar (hidden)
Re-run with --all to show hidden rows
"
`;

exports[`CLI test: mysql2 can diff tables data 1`] = `
"Diff of tables content (rows 1 to 20):

id field_1    updated_at
1  12 → 12.30 [null] → 2020-07-24T18:34:00.000Z
2  30 → 30.45 2020-07-24T18:34:00.000Z → 2020-07-24T19:25:00.000Z

Rows: 2x changed
"
`;

exports[`CLI test: mysql2 can list tables 1`] = `
"table   rows bytes
table_1 2    32.8 kB
table_3 0    32.8 kB
table_2 2    16.4 kB

(81.9 kB in 3 tables)
"
`;

exports[`CLI test: mysql2 can load dump to database 1`] = `
"column     type         nullable default foreign key
id         int unsigned false
field_1    int          true
field_2    varchar(50)  true
created_at datetime     true
updated_at datetime     true

index                 algorithm unique columns
PRIMARY               BTREE     true   id
table_1_field_1_index BTREE     false  field_1
"
`;

exports[`CLI test: mysql2 can load dump to database 2`] = `
"column     type         nullable default foreign key
id         varchar(255) false
field_1    decimal(8,2) false
field_2    text(65535)  true
created_at datetime     true
updated_at datetime     true

index                  algorithm unique columns
PRIMARY                BTREE     true   id
table_2_field_1_unique BTREE     true   field_1
"
`;

exports[`CLI test: mysql2 can load dump to database 3`] = `
"column     type         nullable default           foreign key
field_1    bigint       false
field_2    varchar(255) false
field_3    int unsigned true                       table_1.id
created_at timestamp    false    CURRENT_TIMESTAMP
updated_at timestamp    false    CURRENT_TIMESTAMP

index                   algorithm unique columns
PRIMARY                 BTREE     true   field_1,field_2
table_3_field_3_foreign BTREE     false  field_3
"
`;

exports[`CLI test: mysql2 can see dumpfile contents 1`] = `
"table                        rows bytes
migrations_dump_mysql_2      3    4.1 kB
migrations_dump_mysql_2_lock 1    4.1 kB
table_1                      2    4.1 kB
table_2                      2    4.1 kB
table_3                      0    4.1 kB

(20.5 kB in 5 tables)
"
`;

exports[`CLI test: mysql2 can see dumpfile contents 2`] = `
"column     type        nullable default foreign key
id         integer     false
field_1    integer     true
field_2    varchar(50) true
created_at datetime    true
updated_at datetime    true

index                 algorithm unique columns
table_1_field_1_index unknown   false  field_1
"
`;

exports[`CLI test: mysql2 can show table 1`] = `
"column     type         nullable default foreign key
id         varchar(255) false
field_1    decimal(8,2) false
field_2    text(65535)  true
created_at datetime     true
updated_at datetime     true

index                  algorithm unique columns
PRIMARY                BTREE     true   id
table_2_field_1_unique BTREE     true   field_1
"
`;

exports[`CLI test: mysql2 database dump contents are valid 1`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_1\\", (t) => {
    t.increments(\\"id\\");
    t.integer(\\"field_1\\");
    t.string(\\"field_2\\", 50);
    t.datetime(\\"created_at\\");
    t.datetime(\\"updated_at\\");
    t.index([\\"field_1\\"], \\"table_1_field_1_index\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_1\\");
};
"
`;

exports[`CLI test: mysql2 database dump contents are valid 2`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_2\\", (t) => {
    t.string(\\"id\\", 255).primary();
    t.decimal(\\"field_1\\", 8, 2).notNullable();
    t.text(\\"field_2\\", 65535);
    t.datetime(\\"created_at\\");
    t.datetime(\\"updated_at\\");
    t.unique([\\"field_1\\"], \\"table_2_field_1_unique\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_2\\");
};
"
`;

exports[`CLI test: mysql2 database dump contents are valid 3`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_3\\", (t) => {
    t.bigInteger(\\"field_1\\").notNullable();
    t.string(\\"field_2\\", 255).notNullable();
    t.integer(\\"field_3\\").unsigned().references(\\"table_1.id\\");
    t.timestamp(\\"created_at\\").notNullable().defaultTo(knex.raw(\\"CURRENT_TIMESTAMP\\"));
    t.timestamp(\\"updated_at\\").notNullable().defaultTo(knex.raw(\\"CURRENT_TIMESTAMP\\"));
    t.primary([\\"field_1\\",\\"field_2\\"]);
    t.index([\\"field_3\\"], \\"table_3_field_3_foreign\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_3\\");
};
"
`;

exports[`CLI test: mysql2 database dump contents are valid 4`] = `
"{\\"id\\":1,\\"field_1\\":12,\\"field_2\\":\\"foo\\",\\"created_at\\":\\"2020-07-24T18:34:00.000Z\\"}
{\\"id\\":2,\\"field_1\\":30,\\"field_2\\":\\"bar\\",\\"created_at\\":\\"2020-07-24T18:34:00.000Z\\",\\"updated_at\\":\\"2020-07-24T18:34:00.000Z\\"}
"
`;

exports[`CLI test: mysql2 database dump contents are valid 5`] = `
"{\\"id\\":\\"1\\",\\"field_1\\":\\"12.30\\",\\"field_2\\":\\"foo\\",\\"created_at\\":\\"2020-07-24T18:34:00.000Z\\",\\"updated_at\\":\\"2020-07-24T18:34:00.000Z\\"}
{\\"id\\":\\"2\\",\\"field_1\\":\\"30.45\\",\\"field_2\\":\\"bar\\",\\"created_at\\":\\"2020-07-24T18:34:00.000Z\\",\\"updated_at\\":\\"2020-07-24T19:25:00.000Z\\"}
"
`;
